### PR TITLE
change the components into a functional component

### PR DESCRIPTION
**Pull request by Mhamad-Raad**
# Restructure
## In this branch I have done the followings:
- Restructured the components from class-based to functional-based components.

### CODE REVIEW

File: src/Components/UI/Calculator.js [hunk 1]
```diff
@@ -1,109 +1,108 @@
...
 
-  callReverse = (event) => {
+  const callReverse = (event) => {
     const clickedReverse = event.target.innerText;
-    const result = calculate(this.state, clickedReverse);
-    this.setState(result);
+    const result = calculate({ total, next, operation }, clickedReverse);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
```
Comment (omar-muhamad) [inline src/Components/UI/Calculator.js:55]: - [ ] When trying to do any of the calculator operations nothing happened, So please fix that.

chrome-capture-2022-9-28

**Review by omar-muhamad — CHANGES_REQUESTED** [review]
# Changes Required ♻️
### Hi @Mhamad-Raad,

Nice job, you are almost there.



## (Highlights) Good Points: :+1:
- You documented your work professionally. ✔
- You used the correct workflow.  ✔
- All linters checks are ok. ✔

## (Changes Required) Aspects to improve: ♻️

- [ ] Please check the inline comments below.

## [Optional] suggestions:

- Nothing to mention.

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification**

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

File: src/Components/UI/Calculator.js (hunk 1 above)
Comment (Mhamad-Raad) [reply src/Components/UI/Calculator.js:55]: hi @Omar-Muhamad, can you try again please it is not like that on my computer.

**Review by Munsa1 — APPROVED** [review]
Hi @Mhamad-Raad, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
:heavy_check_mark: Descriptive Readme
:heavy_check_mark: Passing Linters
:heavy_check_mark: Correct Gitflow

## Optional suggestions
- [ ] When the user clicks the `AC` button, and then starts a new calculation, the zero in the display does not disappear, it cuses for a poor user evperience, kindly take a look at it.


_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**